lib/optimizely/decision_service.rb

```diff
@@ -108,7 +107,7 @@ def get_variation(project_config, experiment_key, user_id, attributes = nil, dec
       user_meets_audience_conditions, reasons_received = Audience.user_meets_audience_conditions?(project_config, experiment, attributes, @logger)
       decide_reasons.push(*reasons_received)
       unless user_meets_audience_conditions
-        message = "User '#{user_id}' does not meet the conditions to be in experiment '#{experiment_key}'."
+        message = "User '#{user_id}' does not meet the conditions to be in experiment '#{experiment_id}'."
```

Comment:
keep experiment_key in all messages here and others

Reply:
Yes, Agree! An i think you should keep both id and key in the messages